### DISCUSSION TITLE
Is there a way to create a global variable that would be scoped to a test run?

### DISCUSSION THREAD

**Discussion by gajus** [avajs/ava#2892] [Q&A]
We need to set a global (unique) value for each test run, i.e. unique test execution ID.

I cannot think of what mechanism could work for this.

`gobal.TEST_ID = uuid()` would only work in `--serial` mode.

**Comment by novemberborn** [comment]
If I'm understanding this correctly, you have a test file with multiple `test()` calls, each calling shared code that needs to access a global variable, yet you want that variable to hold a different value for code invoked from separate tests?

This would work with serial tests (`test.serial()`) or synchronous `test()` calls. It becomes impossible for asynchronous tests unless perhaps you use https://nodejs.org/api/async_context.html#class-asynclocalstorage.
